mempool/src/shared_mempool/peer_manager.rs

```diff
@@ -84,6 +110,67 @@ impl PeerManager {
         {
             state.is_alive = false;
         }
+        self.update_failover();
+    }
+
+    fn update_failover(&self) {
```

Comment:
Think it would be good to add a few more comments to this function.  At least a high level comment above it that describes what the overall purpose of the function is and maybe a couple on the loops to describe what they are doing to make it more easily readable